.gitpod.yml

```diff
@@ -0,0 +1,17 @@
+# List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
+tasks:
+  - name: Compile the protocol buffers & launch the backend with docker-compose
+    init: |
+      cd app
+      docker run --rm -w /app -v $(pwd):/app couchers/grpc ./generate_protos.sh
```

Comment:
I noticed that this (and commands before and after) runs in a separate terminal parallel to the "install and launch frontend step". Since the frontend also requires the proto file, it is possible that sometimes the frontend build fails because of the race condition.

Is there a way to make this a required task so that we run this command standalone first, then run the `docker-compose ...` and frontend commands in parallel?

Reply:
I realised that we have commands such as `gp-sync done` and `gp-sync-await` that allow this to happen. My colleague Jan explained this super well in a [community post.](https://community.gitpod.io/t/gitpod-yml-and-making-sure-init-completes-before-command/2818) However, because `docker-compose` is a command that is non-terminating, these commands, unfortunately, did not work. I'll speak with the Gitpod team next week and see if I can log this feature request. I'm sure that this is possible! 🙏🏼

Instead, I opted for `docker-compose` and `yarn install` to run in separate terminals but at the same time. The only "manual" thing developers would have to do is run a `yarn start` in the front-end terminal after the dependencies have been installed and the back-end is up and running in the other terminal.

Reply:
> One other thing I haven't figured out immediately though - is there a way to open the browser again if you closed it by accident?

![CleanShot 2021-08-08 at 16 26 46](https://user-images.githubusercontent.com/17087373/128637255-a67826f0-ab4b-4101-802a-c97df08e685f.png)

Reply:
I don't have that tab on the left!

Reply:
Do you have the sidebar with other tabs/icons at least? If you do you can right click to bring up the context menu and you might be able to reveal it if you click "Remote Explorer"?
![Context menu shown when right clicking on the tab bar in VS Code](https://user-images.githubusercontent.com/13669362/128647372-75814906-cea3-4807-a4f5-c3bfba04389c.png)

Reply:
> The only "manual" thing developers would have to do is run a yarn start in the front-end terminal after the dependencies have been installed and the back-end is up and running in the other terminal.

I think that's fine in my opinion. At least the hard bit of setting up the environment is done so we can add some notes/instructions to let people know they need to do that if they go with the Gitpod route of developing